google/cloud/storage/object_access_control.h

```diff
@@ -0,0 +1,116 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_OBJECT_ACCESS_CONTROL_H_
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_STORAGE_OBJECT_ACCESS_CONTROL_H_
+
+#include "google/cloud/storage/internal/common_metadata.h"
+#include <map>
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+struct ProjectTeam {
+  std::string project_number;
+  std::string team;
+};
+
+/**
+ * A wrapper for the objectAccessControl resource in Google Cloud Storage.
+ *
+ * @see
+ * https://cloud.google.com/storage/docs/json_api/v1/objectAccessControls for
+ * an authoritative source of field definitions.
+ */
+class ObjectAccessControl {
+ public:
+  ObjectAccessControl() = default;
+
+  static ObjectAccessControl ParseFromJson(std::string const& payload);
+
+  //@{
+  /**
+   * @name Well-known values for the role() field..
+   *
+   * The following functions are handy to avoid common typos in the role names.
+   * We use functions instead of enums because enums are not backwards
+   * compatible and are brittle to changes in the server-side.
```

Comment:
I don't follow. What server-side changes would make using the raw enums brittle?

Reply:
Maybe paranoia on my part: I dislike enums because if the server adds a new value then the client library has to be modified to add the new enums.  With strings the library can immediately support the new values, but admittedly you lose the nice functions / constants to avoid typos.